test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -1604,42 +1572,46 @@ def tearDown(self):
                 if jid:
-                    for file in jid:
-                        self.logger.info("found jobdir %s" % file)
-                        jpath = os.path.join(fpath, file)
+                    for files in jid:
+                        self.logger.info("*** found jobdir %s" % files)
+                        jpath = os.path.join(fpath, files)
                         freezer_file = os.path.join(jpath, 'freezer.state')
                         # Thaw the cgroup
                         state = 'THAWED'
                         fn = self.du.create_temp_file(
                             hostname=self.momA, body=state)
-                        ret = self.du.run_copy(hosts=self.momA, src=fn,
-                                               dest=freezer_file, sudo=True,
-                                               uid='root', gid='root',
-                                               mode=0644)
-                        os.remove(fn)
+                        self.du.run_copy(hosts=self.momA, src=fn,
+                                         dest=freezer_file, sudo=True,
+                                         uid='root', gid='root',
+                                         mode=0644)
+                        self.du.rm(hostname=self.momA)
                         cmd = "rmdir " + jpath
                         self.logger.info("deleting jobdir %s" % cmd)
-                        # This is currently blocked due to infrastructure
-                        # self.du.run_cmd(cmd=cmd,sudo=True)
-        """
+                        self.du.run_cmd(cmd=cmd, sudo=True)
         # Remove the jobdir if any under other cgroups
-        cdir = self.paths['cpuset']
-        if os.path.isdir(cdir):
-            cpath = os.path.join(cdir, 'pbspro')
-            if os.path.isdir(cpath):
-                jid = os.listdir(cpath)
-                if jid:
-                    for file in jid:
-                        self.logger.info("Found jobid %s" %file)
-                        jdir = os.path.join(cpath,file)
-                        cmd2 = "rmdir " + jdir
-                        self.logger.info("deleting jobdir %s" %cmd2)
-                        #self.du.run_cmd(cmd=cmd2, sudo=True)
-        """
+        cgroup_subsys = ('cpuset', 'memory')
+        for subsys in cgroup_subsys:
+            self.logger.info("Looking for orphaned jobdir in %s" % subsys)
+            if self.paths[subsys]:
+                cdir = self.paths['cpuset']
+                if os.path.isdir(cdir):
+                    cpath = os.path.join(cdir, 'pbspro')
+                    if os.path.isdir(cpath):
+                        jid = glob(os.path.join(cpath, "*", ""))
+                        if jid:
+                            for files in jid:
+                                self.logger.info("***Found jobid %s" % files)
+                                jdir = os.path.join(cpath, files)
+                                cmd2 = "rmdir " + jdir
```

Comment:
Isn't this supposed to be a list?  ['rmdir', jdir] ?

Reply:
I looked at other tests in the repo and most of them are constructing command as string and not list. Also the documentation for run_cmd says following

cmd (str or None) – the command to run

Reply:
I think we need to change doc, it should be (list or str)
Regarding: str vs list: However run_cmd works with str but list is preferred if you have space in command name or in any of it's argument because if you pass str then run_cmd will simply do str.split() to create list and pass it to subprocess.Popen() and result of str.split() might not be same command what you want to run so it is good idea to pass list instead str.

Reply:
okay thanks Hiren. I will change this. We should better fix the docs then.